Fixed tests and early x flavor

gitneutrinos opened the pull request.

gitneutrinos committed:
Merge pull request #9 from srichers/master

functioning test case for Sherwood

gitneutrinos committed:
Merge pull request #10 from srichers/master

stuff is more disentagled

gitneutrinos committed:
Update StabilityPackage.wl

gitneutrinos committed:
import check

gitneutrinos committed:
Added in basic gershgorin disc functions

gitneutrinos committed:
GDValue,Box Transform fitting, GD test case in runstuff

added in gershgorin things and fixed an error. Runstuff now runs a GDValue verification test.  Added in box transform fit function (generate intial guesses, generate fits for single radius). Changed kgrid bounds, and ktarget uses total potential.

gitneutrinos committed:
fixed a bug in siPotential

needs to know the number of bins n

gitneutrinos committed:
more bugs

all around siPotential.

gitneutrinos committed:
got rid of potential function

siPotential replaces the need for it

gitneutrinos committed:
hs[[9]] no longer exists

removed

gitneutrinos committed:
Run Template, and package updates

Package Changes:

1) Documentation. ?function_name now returns the arguments required.

2) no dh functions for build hamiltonians, get equations, and kadapt.

3) buildkgrid and kadapt now have optional arguments. A manual k target, or manual bounds on the kgrid can be input. Otherwise, it defaults to automatic targetting and bounds.

4) ellispe box fit. Contains a few functions to generate intial guesses, fit a single radius, and then fit the entire profile. Needs testing/debugging.

5) Fixed GD functions. Needs debugging.

6) makeLocalVariables[] function added to import constants into the run notebook, for conversion purposes ect.

7) Fewer exported functions. Several functions are now missing from the private preamble as they are rarely, if ever, called outside of the package.

8) SI Potential function has been pulled out of the buildHamiltonians function. Returns sqrt[2] Gf (total neutrino number + antis).

Changes to run template:
1) starting to add 2 beam check to the accruacy checks block. Need to figure out the artifical frequency and lotosdo values to appropriately sub in for the mus in the self-interaction.  Trying to avoid defining new buildHamiltonian function if possible.

2) Plotting functions block.  Added some automatic plotting routines.

gitneutrinos committed:
Less Functions

Removed the three "no dh" functions and implemented options instead. Calling kAdapt with the option "dh"-> False removes the hamiltonian perturbation.

gitneutrinos committed:
2 beam checks

2 beam check in runtemplate works. Gets correct answer. Some other clean up of the package.

gitneutrinos committed:
Box fit improvements

Removed the "get intial guess" function. Instead, if the radius entered for the fitsingleradius function is the first radial bin, generates the appropriate guesses at that time.

gitneutrinos committed:
x flavor options

Added options to most functions to allow for non-zero x flavors.  Currently does not actually add them into the Hsi, just finds their potential and calculates starting density matricies.

New option patterns for buildkGrid and kAdapt.

Bug still present in kAdapt if "xflavor"-> True. Works if false. unclear where issue lies, as each function called with xflavor-> true works indvidually.

srichers commented on RunStuff.wl:
```diff
@@ -95,3 +95,133 @@ ListLogPlot[{Transpose@{kdebug[[All,2]],kdebug[[All,3]]},OldData},ImageSize-> Sc
 
 
 
+
```
Do we need all of this commented data?

srichers commented on RunStuff Template.m:
```diff
@@ -0,0 +1,165 @@
+(* ::Package:: *)
```
What is the difference between the template and the regular file?

gitneutrinos replied:
Only keep runstufftemplate (this contains tests)

srichers commented on RunStuff Template.nb:
```diff
@@ -0,0 +1,1769 @@
+(* Content-type: application/vnd.wolfram.mathematica *)
```
Should we remove this file from the repository?

srichers commented on StabilityPackage.wl:
```diff
@@ -5,22 +5,37 @@ BeginPackage["StabililtyPackage`"]
+makeLocalVariables::usage=
+"[],Imports the variables definitions used in the package to the local notebook. Useful for analysis"
+boxFit::usage=
+"[infile,species=1,2]. Box fits a file."
```
perhaps mention that "boxfit" applies to fitting ellipses

srichers commented on StabilityPackage.wl:
```diff
@@ -63,13 +82,57 @@ SelectSingleRadius[data_,ri_]:=Association[
 "mids"->data["mids"] (*Cos\[Theta] bin midpoints*)
 ]
 
+Options[siPotential]={"xflavor"-> False};
```
lots of this looks like it is duplicated below. Is there no way to avoid the duplication?

srichers commented on StabilityPackage.wl:
```diff
@@ -63,13 +82,57 @@ SelectSingleRadius[data_,ri_]:=Association[
 9=HsiRadial
 *)
-buildHamiltonians[data_,testE_,hi_]:=Module[{n,\[Theta],name11,name12,name21,name22,\[Rho],\[Rho]b,A,Ab,Hm,Hvac,\[Mu],\[Mu]b,Hsi,H,Hb,\[Delta]H,\[Delta]Hb,nudensity,nubardensity,Ve,\[Omega],HsiRad},(
+Options[buildHamiltonians]={"dh"-> True,"xflavor"-> False};
```
What does dh do? Perhaps rename something like "do_perturb_H"?

srichers replied:
nevermind - see below - I think it is better to not include this functionality for simplicity

srichers commented on StabilityPackage.wl:
```diff
@@ -93,39 +156,50 @@ Ab[i]={{0,ToExpression[StringJoin["Ab",name12,ToString[i]]]},{ToExpression[Strin
 Hb[i]=Hvac-Hm-Hsi[i];
-\[Delta]H[i]=Sum[((D[H[i][[1,2]],\[Rho][j][[1,2]]]) A[j])+((D[H[i][[1,2]],\[Rho]b[j][[1,2]]])Ab[j]),{j,1,n}];
-\[Delta]Hb[i]=Sum[((D[Hb[i][[1,2]],\[Rho][j][[1,2]]])A[j])+((D[Hb[i][[1,2]],\[Rho]b[j][[1,2]]])Ab[j]),{j,1,n}];
+\[Delta]H[i]=dhswitch.Sum[((D[H[i][[1,2]],\[Rho][j][[1,2]]]) A[j])+((D[H[i][[1,2]],\[Rho]b[j][[1,2]]])Ab[j]),{j,1,n}];
```
I think we decided the Hamiltonian perturbation is not required, so we should avoid the extra complexity.

srichers commented on StabilityPackage.wl:
```diff
@@ -193,36 +276,168 @@ Return[as]
 
 
 
+
```
Testing should be done in the test file rather than the package file

srichers commented on StabilityPackage.wl:
```diff
@@ -193,36 +276,168 @@ Return[as]
+		
+]
+
+eBoxFitSingleRadius[data_,ri_,species_,guesses_]:=Module[{ebox,es1box,es2box,es3box,ne,Fe,Pre,ane,aFe,aPre,datasr,br,ei,aei,ag,aag,\[Beta]g,a\[Beta]g,\[Chi]g,a\[Chi]g,foc1234,afoc1234,g0,ag0},
```
I think it would be better to pass this function a dataSingleRadius, rather than the full data object and a radial index.

srichers commented on StabilityPackage.wl:
```diff
@@ -5,22 +5,37 @@ BeginPackage["StabililtyPackage`"]
+	"[file, radius start, radius end, test energy, hierarchy, number of k steps, (optional) manual target k, 
+(optional) k lowwer bound multiple; Default \!\(\*SuperscriptBox[\(10\), \(-3\)]\),(optional) k upper bound Default: 10]. Runs the full routine"
+GDValue::usage=
+	"returns the maximum possible value of the frequency"
```
mention that GD is gershgorin disk

srichers commented on StabilityPackage.wl:
```diff
@@ -5,22 +5,37 @@ BeginPackage["StabililtyPackage`"]
+(optional) k lowwer bound multiple; Default \!\(\*SuperscriptBox[\(10\), \(-3\)]\),(optional) k upper bound Default: 10]. Runs the full routine"
+GDValue::usage=
+	"returns the maximum possible value of the frequency"
+GDdata::usage=
```
If it's exactly the same, I think the filename should reflect that. E.g. kadaptGD[]

srichers reviewed (commented):
I think we should stop and work together to get the PR knocked out before continuing changes, or we are shooting at a moving target. That or isolate untested new stuff on a different branch.

gitneutrinos committed:
Delete RunStuff.wl

gitneutrinos committed:
Delete RunStuff Template.nb

gitneutrinos committed:
ndensity function & box fit intialguess

Added a function which computes the mu matricies for nu, nubar, and nux neutrino flavors.  This is the only place where the "xflavor" option makes an actual change, as whether mux is zero or nonzero propogates through.  The option is still required on most functions.  Seperated out the intial guess for the box fit into ti's own function.

gitneutrinos committed:
Update StabilityPackage.wl

gitneutrinos committed:
Modules, eigenvectors, xflavor debug,  xflavor ellipse fit

Bug issue due to the static nature of block variables. All blocks are now modules, which are more flexible. Slight slow down is possible.

Eigenvectors can now be found with evecscale function.

New option in kadapt. If "eigenvectors"-> True, returns each set of eigenvectors instead of each eigenvalue.

xflavor default has been set to true.

Ellipse fit now fits x neutrinos/anti neutrinos.

gitneutrinos committed:
typo

typo fixed in kAdapt. stability matrix was called with many radius data instead of single radius data.

gitneutrinos committed:
Update StabilityPackage.wl

removed some debugging print statements and debugging output.

gitneutrinos committed:
verification test needed option change

gitneutrinos committed:
updted .gitignore

gitneutrinos committed:
Merge branch 'xflavor' into master
